storage/range.go

```diff
@@ -1663,54 +1663,29 @@ func (r *Range) Snapshot() (raftpb.Snapshot, error) {
 	defer snap.Stop()
 	var snapData proto.RaftSnapshotData
 
-	// Certain keys are treated specially while building the snapshot;
-	// see the switch statement below for details.
-	hardStateKey := engine.RaftHardStateKey(r.Desc().RaftID)
-	appliedKey := engine.RaftAppliedIndexKey(r.Desc().RaftID)
-	descKey := engine.RangeDescriptorKey(r.Desc().StartKey)
-
-	// Save the range metadata while iterating. Use the values read from
-	// the snapshot instead of the members of the Range struct because
-	// they might be changed concurrently.
+	// Read the range metadata from the snapshot instead of the members
+	// of the Range struct because they might be changed concurrently.
 	appliedIndex := uint64(raftInitialLogIndex)
+	v, err := engine.MVCCGet(snap, engine.RaftAppliedIndexKey(r.Desc().RaftID),
+		proto.ZeroTimestamp, false, nil)
```

Comment:
You might as well specify consistent=true here. ZeroTimestamp values are inline and can't have intents. Kind of weird to specify it as false.